components/styled/menu/src/molecules/action-menu.tsx

```diff
@@ -120,7 +119,12 @@ const ActionMenu: FC<IProps> = ({
             </Button>
           )}
           {showMenu && (
-            <StyledCard tabIndex={-1} ref={setPopperElement} style={popper.styles.popper} {...popper.attributes.popper}>
+            <StyledCard
+              tabIndex={-1}
+              ref={setPopperElement}
+              style={{ ...popper.styles.popper, zIndex: 3 }}
```

Comment:
I'm worried this sort of patch is just going to re-create the same problem down the line - just incrementing the index to 3 might work for now but would fail if someone builds the entire table at z-index 4. 
I'd like to find a longer term solution - I think the best strategy I've seen so far is in that article I linked where they set constants for different categories of UI component. I'm sure there are other ideas out there too - maybe it's worth setting up a chat with Jack/Jonny too so we can all agree?